### ISSUE TITLE
MLM for Lode Runner is very bad

### ISSUE THREAD

**Issue by williamsr03** [schrum2/MarioDiffusion#129]
MLM seems to majority of the time predict that a period is the masked element, even if it is not resulting in bad scores below:

![Image](https://github.com/user-attachments/assets/9d64a220-7594-4a62-8aec-dc83e8cbcaf5)

![Image](https://github.com/user-attachments/assets/05d27475-0cf3-43c4-9be5-8c53a6f576fd)

**Comment by williamsr03** [comment]
Ran run diffusion on Lode Runner and it seemed to just produce copies of the levels within the dataset.
Currently have the WGAN model running on Lode Runner so going to see if Lode Runner can be used on the interactive unconditional evolution

**Comment by williamsr03** [comment]
After running the MLM for 25000 epochs, it seems that the training loss is around the same as running it for only 300 based on the pictures below. For the training loss (the bottom one), it looks like the training loss (the blue line) is not there but I believe it is but it is just hidden behind the red line. Long story short, it seems that running the MLM for this long did not change its performance

However, after looking at the outputs (top picture), it seems to no longer be just guessing '.' such as guessing short, few, and ladders. This makes me wonder if it is trying to get better but still needs to just train for longer. Therefore before I leave I started training another MLM but for 100000 epochs overnight to see how it compares

<img width="1888" height="879" alt="Image" src="https://github.com/user-attachments/assets/8ca3d03e-f89e-4263-9470-5c4340eb3dc2" />

<img width="995" height="600" alt="Image" src="https://github.com/user-attachments/assets/773028fc-ddcb-452a-ac0a-f06f99fd8d27" />

<img width="995" height="597" alt="Image" src="https://github.com/user-attachments/assets/f5a89f9c-bf34-4e67-8e6d-7ddd130bca23" />

**Comment by williamsr03** [comment]
Below is the results for the 100000 epochs for the MLM. The predicting and accuracy increased significantly while the loss stayed roughly within the same spot. The question is if the MLM is now overfitted on the Lode Runner information. Regardless, I'm going to start running a diffusion model on this new MLM to see if it gets better results.
 
<img width="1911" height="890" alt="Image" src="https://github.com/user-attachments/assets/9a7f7f32-a0c7-427e-b8c5-d1a79fbe5439" />

<img width="1000" height="600" alt="Image" src="https://github.com/user-attachments/assets/c0d4ded2-6ff8-4bc5-a94c-00defc7a370d" />

<img width="1002" height="600" alt="Image" src="https://github.com/user-attachments/assets/ebad862b-8339-4d06-8ed6-f8e8bb1e9c94" />

**Comment by schrum2** [comment]
These results are good enough for now. We can keep looking for ways to improve Lode Runner performance, but we're about ready to close this issue. Just make sure that the LR readme and batch files use the increased epoch count for MLM training before closing this.

**Comment by williamsr03** [comment]
I believe that I am currently at a good stopping point regarding this issue and Lode Runner in general expect for the diffusion model still running. Even though the current checkpoint has good outputs, being able to see the end would be better. This could definelty be branched out on in the future.
